### PR TITLE
Add weekly scheduled smoke tests

### PR DESCRIPTION
**Pull request by AriPerkkio**
Adds weekly run smoke tests. See https://github.com/sindresorhus/eslint-plugin-unicorn/issues/912#issuecomment-739621530. Note that this PR does not change the existing integration tests. 

**Why:**

Unit tests cover only cases which are expected. The AST produced by Javascript and Typescript can cause very unexpected conditions. These often lead into null pointers and can crash the linter.

ESLint rules should never crash. A crashing rule causes linter to lose all valid linting problems.

**How:**

Example run: https://github.com/AriPerkkio/eslint-plugin-unicorn/actions/runs/405055179
Contains 1832 repositories. A single run takes ~2 hours on Github Actions CI.

Job is scheduled to be run on Sunday 00:00. There is also an option for running this job manually. 

All comments are welcome!

### CODE REVIEW

File: .github/workflows/smoke-test.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 14
+    - run: yarn install
+    - run: yarn link
+    - run: yarn link eslint-plugin-unicorn
+    - run: yarn smoke
+      env:
+        CI: true
```
Comment (fisker) [inline .github/workflows/smoke-test.yml:21]: GitHub Actions set this automaticly

File: .github/workflows/smoke-test.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 14
```
Comment (fisker) [inline .github/workflows/smoke-test.yml:15]:
```suggestion
```

Let's remove this, use the latest LTS

**Review by fisker — CHANGES_REQUESTED** [review]

File: .github/workflows/smoke-test.yml (hunk 2 above)
Comment (AriPerkkio) [reply .github/workflows/smoke-test.yml]: What's the syntax for `LTS`? By removing this the CI will default to some old version.

```
Expected version ">=12.11". Got "10.23.0"
```
https://github.com/AriPerkkio/eslint-plugin-unicorn/runs/1509597287?check_suite_focus=true

File: package.json [hunk 3]
```diff
@@ -96,7 +98,10 @@
 			"plugin:eslint-plugin/all"
 		],
 		"ignores": [
-			"test/integration/{fixtures,unicorn}/**"
+			"test/integration/{fixtures,unicorn}/**",
+			"test/smoke/**",
```
Comment (fisker) [inline package.json:102]: Why are we ignoring this file?

File: .github/workflows/smoke-test.yml (hunk 2 above)
Comment (AriPerkkio) [reply .github/workflows/smoke-test.yml]: This seems to work now with `actions/setup-node@v2-beta`.

Comment (fisker) [reply .github/workflows/smoke-test.yml]: Do we need care about Node.js version?

If not let's use whatever this action decided, so we don't need update it in future.

Comment (AriPerkkio) [reply .github/workflows/smoke-test.yml]: Minimum requirement is `12.11` which introduced some new features to `worker_threads`, required by `eslint-remote-tester`. 

The v14 was originally copied from integration step:

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d669707d18609b9b4672c7aa768076d01d671948/.github/workflows/main.yml#L50-L54

Comment (fisker) [reply .github/workflows/smoke-test.yml]: Okay, add it back.

File: package.json (hunk 3 above)
Comment (AriPerkkio) [reply package.json]: Ignoring removed. Applied required styling to contents of `test/smoke/**`.

File: test/smoke/repositories.json [hunk 4]
```diff
@@ -0,0 +1,1834 @@
+[
+    "AriPerkkio/js-framework-playground",
```
Comment (fisker) [inline test/smoke/repositories.json:2]: I think @sindresorhus prefer to use `Tab`

File: .github/workflows/smoke-test.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    - run: yarn install
+    - run: yarn link
+    - run: yarn link eslint-plugin-unicorn
```
Comment (fisker) [inline .github/workflows/smoke-test.yml:18]: Will `npm link` work? Other CI jobs uses `npm`

Comment (AriPerkkio) [reply .github/workflows/smoke-test.yml:18]: I've found that `yarn link` is extremely faster than `npm link`, the difference is significant. In the workflow we can use `npm install`, but for creating the package link I would recommend to use yarn.

Comment (fisker) [reply .github/workflows/smoke-test.yml:18]: I use `yarn` too, but in this repo seems all commands using `npm`.

Comment (AriPerkkio) [reply .github/workflows/smoke-test.yml]: Now using `npm`. I tested one CI run and it seems to work fine.

**Comment by fisker** [comment]
I wounder if it can open an issue or something to notify maintainers when problem found.

**Comment by AriPerkkio** [comment]
> I wounder if it can open an issue or something to notify maintainers when problem found.

There is an option available in `eslint-remote-tester.config.js` which can be used to implement features like this. https://github.com/AriPerkkio/eslint-remote-tester#eslint-remote-testerconfigjs

```js
    /**
     * Optional callback invoked once scan is complete.
     *
     * @param {{
     *     repository: string,
     *     repositoryOwner: string,
     *     rule: string,
     *     message: string,
     *     path: string,
     *     link: string,
     *     extension: string,
     *     source: string,
     *     error: (string|undefined),
     * }[]} results Results of the scan, if any
     * @returns {Promise<void>|void}
     */
    onComplete: async function onComplete(results) {
        // Extend the process with custom features, e.g. send results to email, create issues to Github...
    },
```

**Review by fisker — APPROVED** [review]
Looks good, can you fix conflicts?

**Comment by AriPerkkio** [comment]
Rebased from upstream master. Should be good to go.

Thanks for the review @fisker.

File: .gitignore [hunk 6]
```diff
@@ -4,3 +4,5 @@ yarn.lock
 coverage
 package-lock.json
 /test/integration/fixtures
+.cache-eslint-remote-tester
+eslint-remote-tester-results
```
Comment (fisker) [inline .gitignore:8]:
```suggestion
eslint-remote-tester-results

```

**Review by sindresorhus — APPROVED** [review]

**Comment by fisker** [comment]
Thanks!

**Pull request merged**